src/main/java/net/coderbot/iris/uniforms/CommonUniforms.java

```diff
@@ -68,17 +70,39 @@ private static float getBlindness() {
 		return 0.0F;
 	}
 
+	private static float getPlayerMood() {
+		if (client.player == null) {
+			return 0.0F;
+		}
+		return client.player.getMoodPercentage();
+	}
+
 	private static float getNightVision() {
 		Entity cameraEntity = client.getCameraEntity();
```

Comment:
Since blindness has the same variable call, possible to put it up top as:
`private static final Entity cameraEntity = client.getCameraEntity();`

Reply:
That could cause weird issues because the camera entity can change over time (such as when you spectate a different entity in spectator mode),  so it shouldn't be treated like a constant. We can assume that the client is constant however.

Reply:
Just added

Reply:
oh oops, didn't get your message

Reply:
I'll rollback